src/DirectionProvider.jsx

```diff
@@ -38,9 +38,9 @@ export default class DirectionProvider extends React.Component {
     };
   }
 
-  componentWillReceiveProps(nextProps) {
```

Comment:
This is still a safe method to use, despite React's deprecations; `componentDidUpdate` ends up happening later than cWRP. I'm not clear on why this change is helpful or desirable.

Reply:
Disregarding that this will break in the future, many warning messages are noisy and unhelpful when debugging or working with code. cWRP does happen before cDU but there's no clear reason the events in this function need to happen before a render update anyway, and the methods used in this PR align with React's recommendations on the matter.

Reply:
The children of DirectionProvider need the direction to be able to render properly; so by setting this earlier, we potentially avoid multiple extra renders.

Reply:
This is tricky. They've seemed to strip all `*will*` lifecycle hooks from "safeness". We could bypass the lifecycle completely and just export the broadcaster explicitly from DirectionProvider to `withDirection` and let the `withDirection` components handle their own updating, then we don't have to worry about repeat lifecycle. This actually seems more reasonable than arbitrarily using `context` to handle the broadcaster (realistically, DirectionProvider should only be used once per app, anyway, so there doesn't seem to be a need to for context, unless I'm misunderstanding.).

Reply:
I have a feeling that the proper solution here is feature-detecting the new React context API, and using that when available - otherwise falling back to the existing implementation, as-is, in master.